core/src/main/java/org/apache/calcite/sql/fun/SqlMonotonicBinaryOperator.java

```diff
@@ -100,21 +101,19 @@ public SqlMonotonicBinaryOperator(
         return mono1;
       }
       assert getName().equals("*");
-      if (!call.isOperandNull(0, true)) {
-        BigDecimal value = call.getOperandLiteralValue(0, BigDecimal.class);
-        switch (value == null ? 1 : value.signum()) {
-        case -1:
-          // negative constant * mono1 --> reverse mono1
-          return mono1.reverse();
-
-        case 0:
-          // 0 * mono1 --> constant (zero)
-          return SqlMonotonicity.CONSTANT;
-
-        default:
-          // positive constant * mono1 --> mono1
-          return mono1;
-        }
```

Comment:
Add a comment to note that operand0 and operand1 will never be null when code goes here.